drivers/hmis_supplemental/app/controllers/hmis_supplemental/data_sets_controller.rb

```diff
@@ -0,0 +1,87 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# Admin data sets
+module HmisSupplemental
+  class DataSetsController < ApplicationController
+    # I guess this is our stand-in perm for "current user is admin"
+    before_action :require_can_edit_users!
+
+    def index
+      @data_sets = data_set_scope
+    end
+
+    def new
+      @data_set = data_set_scope.new(field_config: '[]')
+      @data_set.build_remote_credential
+    end
+
+    def create
+      @data_set = data_set_scope.new(data_set_params)
+      cred = @data_set.build_remote_credential
+      cred.active = true
+      @data_set.attributes = data_set_params
+      if @data_set.save
+        flash[:notice] = 'Data set created'
+        redirect_to action: :index
+      else
+        render :new
+      end
+    end
+
+    def edit
+      @data_set = load_data_set
+    end
+
+    def update
+      @data_set = load_data_set
+      @data_set.attributes = data_set_params
+      if @data_set.save
+        flash[:notice] = "#{@data_set.name} was successfully updated."
+        redirect_to action: :index
+      else
+        render :edit
+      end
+    end
+
+    def destroy
+      @data_set = load_data_set
+      @data_set.destroy!
```

Comment:
Should we offer a means of destroying the underlying fields/data as well?  Maybe deleting a data set should remove the data?  It will be removed from the view, but potentially hangs around indefinitely.

Reply:
right now we prevent it from being destroyed if there are values (restrict_with_exception) but maybe that's too troublesome. I'll change it to delete_all